src/main/java/seedu/address/model/applicant/ApplicantStatus.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.model.applicant;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents an Applicant Status of an applicant in the address book.
+ */
+public class ApplicantStatus {
+    private static final int APPLICANTSTATUS_REJECTED = 0;
+    private static final int APPLICANTSTATUS_ACCEPTED = 1;
+    private static final int APPLICANTSTATUS_PENDING = 2;
+
+    private static final String[] STATUSES = {"rejected", "accepted", "pending"};
+
+    public final Integer applicantStatus;
+
+    /**
+     * Constructs an Applicant Status
+     *
+     * @param status An applicant's job application status
+     */
+    public ApplicantStatus(int status) {
+        requireNonNull(status);
+        applicantStatus = status;
+    }
```

Comment:
Maybe add an isValidStatus to check if the input int is within the right range? E.g is 0,1 or 2 else throw exception